### ISSUE TITLE
axios.min.js includes unnecessary package.json since 0.21.2

### ISSUE THREAD

**Issue by kawanet** [axios/axios#4028]
#### Describe the bug

`dist/axios.min.js` of version 0.21.2 to 0.21.4 contains its `package.json` which is never required for usages on browser.
Version 0.21.1 and before did not. This gave extra 2KB fat for the minified build.

#### To Reproduce

```
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.4/axios.min.js | wc -c
   19157
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.2/axios.min.js | wc -c
   19025
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.1/axios.min.js | wc -c
   14355

$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.4/axios.min.js | grep -c '"name":"axios","version":"0.21'
1
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.2/axios.min.js | grep -c '"name":"axios","version":"0.21'
1
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.1/axios.min.js | grep -c '"name":"axios","version":"0.21'
0
```

#### Expected behavior

Please keep `dist/axios.min.js` weight healthy. 💪

#### Environment

 - Axios Version 0.21.2 to 0.21.4

#### Additional context/Screenshots

This happens because [lib/helpers/validator.js](https://github.com/axios/axios/commit/5ad6994da#diff-a49cff66d6cc35c2eed6284efa2fd160f18c710edfbf0d7d48046b9cf5624f83R3) loads `package.json` even for `dist/axios.min.js` build. @DigitalBrainJS 

```diff
var pkg = require('./../../package.json');
var currentVerArr = pkg.version.split('.');
```

**Comment by DigitalBrainJS** [comment]
Thanks for reporting! I'll be looking for a workaround to inline a version without importing all the JSON.

**Comment by jasonsaayman** [comment]
Ok cool @DigitalBrainJS please ping me when you are have something thanks

**Comment by DigitalBrainJS** [comment]
@jasonsaayman Done #4029

**Comment by kawanet** [comment]
@DigitalBrainJS 

Thank you for your lightning work at [2e4594a08](https://github.com/DigitalBrainJS/axios/commit/2e4594a08)! It'd shape `axios.min.js` minified.

```diff
- var pkg = require('./../../package.json');
+ var VERSION = require('../env/data.json').version;
```

**Suggestion 1**
[lib/adapters/http.js](https://github.com/axios/axios/blob/master/lib/adapters/http.js) also loads `package.json` as well.
Browser version of `axios.min.js` would not contain `lib/adapters/http.js`, however.
For **consistency**, why don't `lib/adapters/http.js` load the new `env/data.json` instead of `package.json`.

```js
var pkg = require('./../../package.json');
```

**Suggestion 2**
0.21.2 was the very first time for browser version of axios to load a file type of `.json` but not `.js`.
Some developers are claiming that loading `.json` is causing some troubles when bundling with webpack.
Why don't load `env/data.js` instead of `env/data.json` then?
If you prefer to load a `.json` file type directly, such change should have been done at the next major/minor updated version number not at [a patch number](https://semver.org/).

```diff
- { "version": "0.21.5" }
+ exports.version = "0.21.5";
```

**Comment by jasonsaayman** [comment]
@DigitalBrainJS I like the idea that @kawanet has, I can always manually update this file on release, I need to anyways do it in several places.

**Comment by Liwoj** [comment]
@kavanet 
*Some developers are claiming that loading .json is causing some troubles when bundling with webpack.* - this is pretty vague. I bet most of the mentioned "problems" are in fact same problem as here - that whole JSON file gets included when imported into a app...

@DigitalBrainJS 
There is a alternative (and recommended by Webpack devs) way how to include version field from `package.json` and that the use of DefinePlugin - see [this issue](https://github.com/webpack/webpack/issues/237)

**Comment by kawanet** [comment]
@Liwoj Thank you for your comment!
I understand that loading `data.json` via webpack would not cause a trouble any more.
I'm not a webpack user, but a browserify lover in real.
Please ignore suggestion 2 above because browserify would have no trouble on loading `data.json` as well. ✌

**Comment by DigitalBrainJS** [comment]
@kawanet 
1) I've only optimized browser build, and as you said since the builders don't load the `http` adapter for browser builds, it doesn't make much sense to load `env / data.json`. But we can do this anyway in order to have one shared configuration.
2) Yes, I thought about JS config file, but JSON format is more natural for [generated] configs, moreover, it's a bit simpler to write JSON data to file from the code. Yes, webpack has a strange JSON loader, unlike rollup that imports JSON as an object without JSON.parse in the generated code. All popular builders work fine with JSON. Anyway, I'm going to refactor the code for using js just due to the ugly JSON.parse generating by webpack for JSON imports.

**Comment by DigitalBrainJS** [comment]
@Liwoj `There is a alternative`
Not really. Yes, it's good to inline variables and it's an acceptable solution to build minified version of Axios. But what if the user will import Axios package directly from his own code and then build a project using webpack or rollup? Maybe I'm wrong, but I don't see any chance that bundlers will be able to use the already minified package version, so they will import the code from the main entry point, which imports all the package.json to load one variable as there is no any minified entry point in package.json.

**Comment by DigitalBrainJS** [comment]
@jasonsaayman Done.

**Comment by Liwoj** [comment]
@DigitalBrainJS You are right. It is great solution for normal apps but not for libs as Axios. My bad...

**Comment by jasonsaayman** [comment]
I have released 0.22.0 please check if that fixes your error, if not please open a new issue :)

**Comment by kawanet** [comment]
confirmed: `e.exports = {version: "0.22.0"}` without other props.
It looks 1,341 bytes smallified than the previous build then.

```
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.22.0/axios.min.js | wc -c
   17816
$ curl -s https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.4/axios.min.js | wc -c
   19157
```

Thank you for maintaining the great library!
